### PR TITLE
Slimming down Javascript bundle

### PR DESCRIPTION
**Pull request by mattolson**
#### Responding to code review on https://github.com/bigcommerce/cornerstone/pull/1229 that was done post-merge

Based on recommendations from @davidchin in https://github.com/bigcommerce/cornerstone/pull/1229#discussion_r : 
* Move stencil-cli and grunt-cli to devDependencies
* Use npx to run tests using local instead of global packages
* Add Node 8.x (can't support 10 yet because stencil-cli does not)

Based on https://github.com/bigcommerce/cornerstone/pull/1229#discussion_r189137920 :
* Change function passed to jQuery each so reference to `this` is correct

Based on https://github.com/bigcommerce/cornerstone/pull/1229#discussion_r189139469 : 
* Add webpack performance warnings for large assets and chunks

Based on https://github.com/bigcommerce/cornerstone/pull/1229#discussion_r189140458 : 
* Enable source maps in production

#### Update to babel 7

* Remove babel-polyfill and instead rely on useBuiltIns: 'usage'
* Use @babel/preset-env
* Set targets
* Add webpack-bundle-analyzer to help us visualize webpack bundles
* Remove transform-regenerator plugin
* Bump versions to latest for all babel related packages
* Use minimized jquery from distribution
* Use splitChunks: all to separate node_modules into a separate file
  to avoid re-downloading it again when minor changes are made to app.js

#### Reduce usage of sweetAlert and compareProducts
* Remove sweetalert2 from global bundle
* Create Foundation reveal modal to be used for error messages
* Replace usage of sweetalert2 with alert modal in every place
  except cart, shipping estimator, and account details, which
  have more complex use cases
* Change global/sweet-alert to return the swal instance so we
  can call it anywhere to get the initialized instance, rather than
  doing this in jquery onReady
* Bump sweetalert to latest 6.x
* Remove compareProducts from global bundle and add to category,
  brand, compare, and search pages.

#### Remove async/await
* Removal of async/await results in bundle savings because webpack
  doesn't need to include a shim

#### Switch from pace to nanobar
* Pace is an unmaintained, large library that we use for the loading progress indicator. Replace with nanobar and jQuery ajax hooks.

## Impact

The bundle size for the entrypoint has been reduced from 376kb to 286kb.

#### Before
```
[matt.olson@C02X425EJG5M:~/workspace/cornerstone] $ URL=http://matt-olson.mybigcommerce.com npm run lighthouse

> bigcommerce-cornerstone@2.6.0 lighthouse /Users/matt.olson/workspace/cornerstone
> npx lighthouse --output json --chrome-flags="--headless" --quiet --disable-device-emulation --throttling-method=provided $URL | jq '{ttfb: .audits | .["time-to-first-byte"] | .rawValue, paint: .audits | .["first-meaningful-paint"] | .rawValue, interactive: .audits.interactive.rawValue}'

{
  "ttfb": 277.21999999999997,
  "paint": 702.078,
  "interactive": 845.0290001220703
}
```

#### After
```
[matt.olson@C02X425EJG5M:~/workspace/cornerstone] $ URL=http://matt-olson.mybigcommerce.com npm run lighthouse

> bigcommerce-cornerstone@2.6.0 lighthouse /Users/matt.olson/workspace/cornerstone
> npx lighthouse --output json --chrome-flags="--headless" --quiet --disable-device-emulation --throttling-method=provided $URL | jq '{ttfb: .audits | .["time-to-first-byte"] | .rawValue, paint: .audits | .["first-meaningful-paint"] | .rawValue, interactive: .audits.interactive.rawValue}'

{
  "ttfb": 245.334,
  "paint": 523.496,
  "interactive": 618.681
}
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-  - bundle install
-  - npm install -g grunt-cli
-  - npm install -g bigcommerce/stencil-cli
+  - bundle install --path vendor/bundle
   - npm install
 script:
-  - grunt check
-  - stencil bundle
+  - npx grunt check
+  - npx stencil bundle
```
Comment (junedkazi) [inline .travis.yml]: why do we need to run stencil bundle on travis ?

File: assets/js/theme/common/product-details.js [hunk 2]
```diff
@@ -370,10 +368,7 @@ export default class ProductDetails {
                 const tmp = document.createElement('DIV');
                 tmp.innerHTML = errorMessage;
 
-                return swal({
-                    text: tmp.textContent || tmp.innerText,
-                    type: 'error',
-                });
+                return showAlertModal(tmp.textContent || tmp.innerText);
```
Comment (junedkazi) [inline assets/js/theme/common/product-details.js]: Can you please add a before and after for this change

File: package.json [hunk 3]
```diff
@@ -7,66 +7,64 @@
...
+    "webpack-bundle-analyzer": "^3.0.3",
+    "webpack-cli": "^3.1.2",
     "webpack-merge": "^4.1.2"
   },
   "scripts": {
-    "build": "webpack --config webpack.prod.js",
-    "buildDev": "webpack --config webpack.dev.js"
+    "build": "npx webpack --config webpack.prod.js",
+    "buildDev": "npx webpack --config webpack.dev.js",
+    "lighthouse": "npx lighthouse --output json --chrome-flags=\"--headless\" --quiet --disable-device-emulation --throttling-method=provided $URL | jq '{ttfb: .audits | .[\"time-to-first-byte\"] | .rawValue, paint: .audits | .[\"first-meaningful-paint\"] | .rawValue, interactive: .audits.interactive.rawValue}'"
```
Comment (junedkazi) [inline package.json]: It would be pretty cool if we could surface this information on the PR as a comment.

File: package.json [hunk 4]
```diff
@@ -7,66 +7,64 @@
...
-    "babel-core": "^6.26.0",
-    "babel-eslint": "^8.2.2",
-    "babel-loader": "^7.1.4",
-    "babel-plugin-dynamic-import-webpack": "^1.0.2",
-    "babel-plugin-lodash": "^3.3.2",
-    "babel-plugin-syntax-dynamic-import": "^6.18.0",
-    "babel-plugin-transform-regenerator": "^6.26.0",
-    "babel-plugin-transform-runtime": "^6.23.0",
-    "babel-preset-env": "^1.6.1",
+    "@bigcommerce/stencil-cli": "^1.15.4",
```
Comment (junedkazi) [inline package.json]: why do we need this ?

File: .travis.yml (hunk 1 above)
Comment (junedkazi) [reply .travis.yml]: Yes, right now we just run it. But it does nothing with the result. Also travis will still success even if it fails. So I am not sure how it helps. Should we just pull it out for now till we figure out how we can test bundles ?

Comment (mattolson) [reply .travis.yml]: I had assumed that stencil bundle would return a non-zero status code if it fails to bundle, and if so, it should fail the Travis script. Is that not true?

Comment (junedkazi) [reply .travis.yml]: tbh I have never ever seen it fail on stencil bundle 🤷‍♂️

**Comment by mattolson** [comment]
@Ubersmake This is good to go from my perspective. Can you help with some additional testing? It's a pretty broad change, so extra eyes will be helpful.

File: webpack.common.js [hunk 5]
```diff
@@ -19,46 +20,51 @@ module.exports = {
...
+            openAnalyzer: false,
+        }),
     ],
     resolve: {
         alias: {
-            'jquery-migrate': path.resolve(__dirname, 'node_modules/jquery-migrate/dist/jquery-migrate.min.js'),
+            jquery: path.resolve(__dirname, 'node_modules/jquery/dist/jquery.min.js'),
             jstree: path.resolve(__dirname, 'node_modules/jstree/dist/jstree.min.js'),
             lazysizes: path.resolve(__dirname, 'node_modules/lazysizes/lazysizes.min.js'),
             pace: path.resolve(__dirname, 'node_modules/pace/pace.min.js'),
```
Comment (sacr3dc0w) [inline webpack.common.js]: Should swap the Pace alias for Nanobar.

`nanobar: path.resolve(__dirname, 'node_modules/nanobar/nanobar.min.js'),`

Comment (mattolson) [reply webpack.common.js]: Good catch, thanks!

File: templates/components/common/body.html [hunk 6]
```diff
@@ -4,4 +4,5 @@
         {{#block "page"}} {{/block}}
     </div>
     {{> components/common/modal}}
+    {{> components/common/alert-modal}}
```
Comment (davidwrpayne) [inline templates/components/common/body.html]: I guess we still need the regular modal too?

Comment (mattolson) [reply templates/components/common/body.html]: Yeah, the regular modal is used for things like the product quick view. Alert modal is more specialized.

**Pull request merged**